### PR TITLE
coprocessor/*:use structured log format

### PR DESCRIPTION
**Pull request by AndreMouche**
Signed-off-by: AndreMouche <AndreMouche@126.com>

## What have you changed? (mandatory)

Update the log format according to the RFCs.

Check the document for how the format work: https://docs.rs/slog/2.0.0-3.1/slog/macro.log.html#values-implementing-fmtdisplay

## What are the type of changes? (mandatory)

The currently defined types are listed below, please pick one of the types for this PR by removing the others:
- Improvement (a non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

Unit test

## Does this PR affect documentation (docs) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

## Refer to a related PR or issue link (optional)

https://github.com/tikv/rfcs/blob/master/text/2018-12-19-unified-log-format.md
https://github.com/tikv/rfcs/blob/master/text/2018-11-12-unified-key-format.md

### CODE REVIEW

**Comment by AndreMouche** [comment]
@breeswish @huachaohuang PTAL.

File: src/coprocessor/codec/convert.rs [hunk 1]
```diff
@@ -185,12 +185,16 @@ fn bytes_to_f64_without_context(bytes: &[u8]) -> Result<f64> {
         Ok(s) => match s.trim().parse::<f64>() {
             Ok(f) => f,
             Err(e) => {
-                error!("failed to parse float from {}: {}", s, e);
+                error!("failed to parse float";
+                "from"=>s,
```
Comment (breeswish) [inline src/coprocessor/codec/convert.rs]:
```
"from" => s
      ^  ^
      space
```

File: src/coprocessor/dag/expr/mod.rs [hunk 2]
```diff
@@ -235,7 +235,8 @@ impl Expression {
     }
 
     pub fn build(ctx: &EvalContext, mut expr: Expr) -> Result<Self> {
-        debug!("build expr:{:?}", expr);
+        debug!("build-expr";
```
Comment (breeswish) [inline src/coprocessor/dag/expr/mod.rs]: maybe we can remove this!

File: src/coprocessor/endpoint.rs [hunk 3]
```diff
@@ -487,7 +487,7 @@ fn make_tag(is_table_scan: bool) -> &'static str {
 }
 
 fn make_error_response(e: Error) -> coppb::Response {
-    error!("{:?}", e);
+    error!("error-response";"error"=>e.to_string());
```
Comment (breeswish) [inline src/coprocessor/endpoint.rs]: Can we use `"error" => %e`?

File: src/coprocessor/dag/expr/mod.rs (hunk 2 above)
Comment (AndreMouche) [reply src/coprocessor/dag/expr/mod.rs]: Seems we cannot.

File: src/coprocessor/codec/convert.rs [hunk 4]
```diff
@@ -186,15 +186,15 @@ fn bytes_to_f64_without_context(bytes: &[u8]) -> Result<f64> {
...
+                "from" => s,
+                "error" => %e,
                 );
                 0.0
             }
         },
         Err(e) => {
             error!("failed to convert bytes to str";
-             "error"=> e.to_string());
+             "error"=> %e);
```
Comment (breeswish) [inline src/coprocessor/codec/convert.rs]:
```
"error" => %e
       ^
```

**Comment by AndreMouche** [comment]
@huachaohuang PTAL

**Comment by AndreMouche** [comment]
@huachaohuang PTAL

File: src/coprocessor/dag/expr/mod.rs [hunk 5]
```diff
@@ -235,7 +235,10 @@ impl Expression {
     }
 
     pub fn build(ctx: &EvalContext, mut expr: Expr) -> Result<Self> {
-        debug!("build expr:{:?}", expr);
+        debug!(
+            "build-expr";
+            "expr" => format!("{:?}", expr)
```
Comment (huachaohuang) [inline src/coprocessor/dag/expr/mod.rs]: Can we use `"expr" => ?expr`?

File: src/coprocessor/local_metrics.rs [hunk 6]
```diff
@@ -50,7 +50,10 @@ impl CopFlowStatistics {
         if let Err(e) = self.sender.schedule(PdTask::ReadStats {
             read_stats: to_send_stats,
         }) {
-            error!("send coprocessor statistics: {:?}", e);
+            error!(
+                "send coprocessor statistics failed";
+                "err" => e.to_string()
```
Comment (huachaohuang) [inline src/coprocessor/local_metrics.rs]: Can we use `"err" => %e`?

**Pull request merged**